### PR TITLE
Mongo config

### PR DESCRIPTION
**Pull request by maleck13**
@philbrookes  Add a data-mongo-replica template based on the 3 node replica set

Can be test by deploying the latest negotiator 0.0.6

then doing the following request to set up the replica set

```
curl -XPOST http://NEGOTIATOR_HOST/service/deploy/data-mongo-replica/ENV -d '{"serviceName":"data-mongo","domain":"rhmap", "replicas":1,"target":{"host":"YOUR_OSCP","token":"YOUR_TOKEN}}'
```

### CODE REVIEW

File: pkg/deploy/configuration.go [hunk 1]
```diff
@@ -245,7 +246,17 @@ func (d *DataMongoConfigure) Configure(client Client, deployment *dc.DeploymentC
 		}
 		return url
 	}
-	dataDc, err := client.FindDeploymentConfigsByLabel(namespace, map[string]string{"rhmap/name": "data"})
+	// look for the Job if it already exists no need to run it again
+	existingJob, err := client.FindJobByName(namespace, deployment.Name+"-dataconfig-job")
+	if err != nil {
+		d.statusUpdate(deployment.Name, "error finding existing Job "+err.Error(), "error")
```
Comment (philbrookes) [inline pkg/deploy/configuration.go]: Should this be an error, that seems a bit alarming. Could it not say "could not find existing Job, will launch new one" or something like that?

File: pkg/mock/paasClient.go [hunk 2]
```diff
@@ -38,6 +38,22 @@ type PassClient struct {
 	Error   map[string]error
 }
 
+func (pc PassClient) CreatePod(ns string, p *api.Pod) (*api.Pod, error) {
```
Comment (philbrookes) [inline pkg/mock/paasClient.go]: This should have a comment

Comment (philbrookes) [inline pkg/mock/paasClient.go]: oh it's in the mock, nvm!

File: pkg/openshift/template.go [hunk 3]
```diff
@@ -25,25 +25,33 @@ const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456
...
 
 	var t = template.New("")
+	var genPass = func() string {
+		b := make([]byte, 16)
+		for i := range b {
+			b[i] = letterBytes[rand.Intn(len(letterBytes))]
+		}
+		return string(b)
+	}
+	var generatedPass = genPass() // a single generated password per template load
```
Comment (philbrookes) [inline pkg/openshift/template.go]: Can we allow for more than one if required? I could imagine a service needing a user and an admin password.

Comment (philbrookes) [inline pkg/openshift/template.go]: oh I see you've done that after all, haha

File: pkg/deploy/configuration.go (hunk 1 above)
Comment (maleck13) [inline pkg/deploy/configuration.go]: This is if it failed try to even find the job (as in unexpectedly failed). If there is no Job it just returns nil

**Pull request merged**